src/utils/fetch-event-data.js

```diff
@@ -0,0 +1,33 @@
+export const EVENTS_URL =
+    'https://www.mongodb.com/api/event/all/1?sort=-node_type_attributes.event_start';
+
+export const removePastEvents = events => {
+    const today = new Date();
+    return events.filter(
+        e =>
+            new Date(e.node_type_attributes.event_end) >= today &&
+            e.status === 'published'
+    );
+};
+
+// Fetches data from mongodb.com/events api
+const fetchEventData = async () => {
+    try {
+        const data = await fetch(EVENTS_URL, {
+            credentials: 'same-origin',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+        });
+        if (data) {
+            const parsedData = await data.json();
+            const upcomingEvents = removePastEvents(parsedData);
+            return upcomingEvents;
+        }
+    } catch (e) {
```

Comment:
One other piece I added in here was to make these fetch calls more robust, so still returning some array even if it errors out. This allows us to still display some events if the other source goes down or is inaccessible.